tesseract_core/sdk/engine.py

```diff
@@ -118,22 +120,22 @@ def wrapper_needs_docker(*args: Any, **kwargs: Any) -> None:
     return wrapper_needs_docker
 
 
-def get_free_port(within_range: tuple[int, int] | None = None) -> int:
+def get_free_port(
+    within_range: tuple[int, int] | None = (49152, 65535),
+    exclude: Sequence[int] = (),
+) -> int:
     """Find a random free port to use for HTTP."""
-    if within_range is None:
-        # Let OS pick a random free port
-        with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as s:
-            s.bind(("localhost", 0))
-            return s.getsockname()[1]
-
     start, end = within_range
-    if start < 0 or end > 65535 or start >= end:
-        raise ValueError("Invalid port range")
+    if start < 1024 or end > 65535 or start >= end:
+        raise ValueError("Invalid port range, must be between 1024 and 65535")
```

Comment:
Why not allow ports < 1024? Those are reserved for system services on Linux, but you can use them if you `sudo` up.

Reply:
w.r.t. the port ranges topic, see also this comment https://github.com/pasteurlabs/tesseract-core/pull/161#discussion_r2086318904

Reply:
That is, we now allow any valid port (0 to 65535) but default to the IANA recs.